### PR TITLE
#105 MVVM architecture for supervised group 

### PR DESCRIPTION
**Pull request by ShawnGuerra-Bautista**
**Changes in this PR:**
https://github.com/Landry333/Big-Owl/pull/135/files/12f958e8baf70572c3bf6d53f9d3e8563d256629..fff45da284465722e7235430f7e094b756dcbfcd

----

Issues: #105 

We basically added the MVVM architecture for supervised group class.

### CODE REVIEW

File: app/src/main/java/com/example/bigowlapp/viewModel/SupervisedGroupListViewModel.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import androidx.lifecycle.MutableLiveData;
+
+import com.example.bigowlapp.model.Group;
+import com.example.bigowlapp.repository.AuthRepository;
+import com.example.bigowlapp.repository.GroupRepository;
+
+import java.util.List;
+
+public class SupervisedGroupListViewModel {
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/viewModel/SupervisedGroupListViewModel.java:11]: Should extend ViewModel

File: app/src/main/java/com/example/bigowlapp/activity/SupervisedGroupListActivity.java [hunk 2]
```diff
@@ -2,65 +2,55 @@
...
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_supervised_group_list);
+        supervisedGroupListViewModel = new SupervisedGroupListViewModel();
         initialize();
     }
 
     protected void initialize() {
         try {
             FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser();
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/SupervisedGroupListActivity.java:37]: Should probably use AuthRepo to get the user, vs FirebaseAuth

**Review by refatK — CHANGES_REQUESTED** [review]
Just a couple of things

**Comment by ShawnGuerra-Bautista** [comment]
> Just a couple of things

@refatK I've completed all your requests.

File: app/src/main/java/com/example/bigowlapp/viewModel/SupervisedGroupListViewModel.java [hunk 3]
```diff
@@ -19,11 +21,19 @@ public SupervisedGroupListViewModel() {
         groupRepository = new GroupRepository();
     }
 
-    public MutableLiveData<List<Group>> setListOfDocumentByArrayContains() {
-        groupLiveData = groupRepository.getListOfDocumentByArrayContains("supervisedUserId",
-                authRepository.getCurrentUser().getUid(), Group.class);
+    public MutableLiveData<List<Group>> getSupervisedGroupList() {
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/viewModel/SupervisedGroupListViewModel.java:24]: Super small thing, but should return this as LiveData not MutableLiveData

**Review by refatK — CHANGES_REQUESTED** [review]
Thanks for the fixes, one more thing :)

**Review by KhangZhengNg — APPROVED** [review]
Overall good job working on MVVM architecture along with @refatK . All methods are very handy and helpful for development.

**Review by darkheros12 — DISMISSED** [review]
Please pull from staging before PR. Once the pull is done, this PR can be approved

File: app/src/main/java/com/example/bigowlapp/viewModel/SupervisedGroupListViewModel.java (hunk 3 above)
Comment (ShawnGuerra-Bautista) [reply app/src/main/java/com/example/bigowlapp/viewModel/SupervisedGroupListViewModel.java]: Makes sense. Consider it done.

**Review by refatK — APPROVED** [review]
Thanks for addressing all the changes i asked for!

**Comment by ShawnGuerra-Bautista** [comment]
CI builds and test passed, but Codecov complains because the coverage isn't improving.

Tests will be pushed in another PR.

**Pull request merged**